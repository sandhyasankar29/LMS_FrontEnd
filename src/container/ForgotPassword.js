import { React, useState } from 'react';
import axios from 'axios';
import {
    Box,
    Button,
    Grommet,
    TextInput,
    Image,
    Select,
} from 'grommet';
import Swal from 'sweetalert2'
import passwordImage from '../assets/password.svg'
import { grommet } from 'grommet/themes';
import { urlPass } from './Url'
const ForgotPassword = () => {
    const Toast = Swal.mixin({
        toast: true,
        position: 'top-end',
        showConfirmButton: false,
        timer: 3000,
        timerProgressBar: true,
        didOpen: (toast) => {
            toast.addEventListener('mouseenter', Swal.stopTimer)
            toast.addEventListener('mouseleave', Swal.resumeTimer)
        }
    })
    const [email, setEmail] = useState("")
    const [id, setId] = useState("")
    const options = ['student', 'teacher']
    const onSubmitForgotPassword = () => {
        if (email) {
            axios.post(urlPass, { email: email, id: id })
                .then(res => {
                    Toast.fire({
                        icon: 'success',
                        title: 'Please Check your Email'
                    })
                }).catch(error => {
                    Toast.fire({
                        icon: 'error',
                        title: 'Failed Try Again'
                    })
                })
        }
    }
    document.body.style.overflow = "hidden"
    return (
        <Grommet full theme={grommet}>
            <Box fill align="center" justify="center">
                <Box height="small" width="large" style={{
                    marginBottom: "3%",
                }}>
                    <Image src={passwordImage} fit="contain" />
                </Box>

                <Box width="medium">
                    <Box
                        width="medium"
                        direction="row"
                        align="center"
                        round="small"
                        border
                    >
                        <TextInput
                            plain
                            placeholder="Enter Registered Email"
                            name="email" type="email"
                            //eslint-disable-next-line react/jsx-no-duplicate-props
                            type={'text'}
                            value={email}
                            onChange={(event) => setEmail(event.target.value)}
                        />
                        <Select
                            id="select"
                            name="id"
                            placeholder="Choose.."
                            value={id}
                            options={options}
                            onChange={({ option }) => setId(option)}
                        />
                    </Box>

                    <Box direction="row" justify="center" margin={{ top: 'medium' }}>
                        <Button active={true} onClick={onSubmitForgotPassword} type="submit" label="Get Password" primary />
                    </Box>
                </Box>
            </Box>
        </Grommet>
    );
};

export default ForgotPassword;