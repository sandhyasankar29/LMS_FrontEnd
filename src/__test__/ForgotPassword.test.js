import React from 'react';
import ReactDOM from 'react-dom';
import ForgotPassword from '../container/ForgotPassword';
import { cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'

afterEach(cleanup)
it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<ForgotPassword></ForgotPassword>, div)
})

it("matches snapshot", () => {
    const tree = renderer.create(<ForgotPassword></ForgotPassword>).toJSON();
    expect(tree).toMatchSnapshot();
})
