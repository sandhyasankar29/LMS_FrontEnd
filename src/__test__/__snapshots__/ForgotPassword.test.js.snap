// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 bWDPZI"
>
  <div
    className="StyledBox-sc-13pk1d4-0 bxevCE"
  >
    <div
      className="StyledBox-sc-13pk1d4-0 bJtzEt"
      style={
        Object {
          "marginBottom": "3%",
        }
      }
    >
      <img
        className="StyledImage-ey4zx9-0 iBeQsm"
        src="password.svg"
      />
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 gfocoz"
    >
      <div
        className="StyledBox-sc-13pk1d4-0 clcAnq"
      >
        <div
          className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 dpaJQp"
        >
          <input
            autoComplete="off"
            className="StyledTextInput-sc-1x30a0s-0 hubMyb"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Enter Registered Email"
            type="text"
            value=""
          />
        </div>
        <button
          aria-label="Open Drop"
          className="StyledButton-sc-323bzc-0 ebqXLG Select__StyledSelectDropButton-sc-17idtfo-1 iGEQF"
          id="select"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="button"
        >
          <div
            className="StyledBox-sc-13pk1d4-0 jmuHez"
          >
            <div
              className="StyledBox-sc-13pk1d4-0 iAryDB"
            >
              <div
                className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 dpaJQp"
              >
                <input
                  autoComplete="off"
                  className="StyledTextInput-sc-1x30a0s-0 iuIygK Select__SelectTextInput-sc-17idtfo-0 bIurki"
                  id="select__input"
                  name="id"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Choose.."
                  readOnly={true}
                  tabIndex="-1"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              className="StyledBox-sc-13pk1d4-0 eA-DBEo"
              style={
                Object {
                  "minWidth": "auto",
                }
              }
            >
              <svg
                aria-label="FormDown"
                className="StyledIcon-ofa7kd-0 fuviCN"
                viewBox="0 0 24 24"
              >
                <polyline
                  fill="none"
                  points="18 9 12 15 6 9"
                  stroke="#000"
                  strokeWidth="2"
                />
              </svg>
            </div>
          </div>
        </button>
      </div>
      <div
        className="StyledBox-sc-13pk1d4-0 cgsQvy"
      >
        <button
          className="StyledButton-sc-323bzc-0 iVPikU"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="submit"
        >
          Get Password
        </button>
      </div>
    </div>
  </div>
</div>
`;
